pkg/anonymization/anonymizer.go

```diff
@@ -231,6 +246,45 @@ func (anonymizer *Anonymizer) ObfuscateIP(ipStr string) string {
 	return "::"
 }
 
+// Stores the translation table in a Secret in the openshift-insights namespace.
+// The actual data is stored in the StringData protion of the Secret.
+// Uses the Apply mechanism so it creates the secret if not present or updates it when present.
+func (anonymizer *Anonymizer) StoreTranslationTable() *corev1.Secret {
+	if len(anonymizer.translationTable) == 0 {
+		return nil
+	}
+	defer anonymizer.ResetTranslationTable()
+
+	secret := &applycorev1.SecretApplyConfiguration{
+		TypeMetaApplyConfiguration: applymetav1.TypeMetaApplyConfiguration{
+			Kind:       &secretKind,
+			APIVersion: &secretAPIVersion,
+		},
+		ObjectMetaApplyConfiguration: &applymetav1.ObjectMetaApplyConfiguration{
+			Name: &TranslationTableSecretName,
+		},
+		StringData: anonymizer.translationTable,
+	}
+
+	applyOptions := metav1.ApplyOptions{
+		Force:        true,
+		FieldManager: "insights-operator",
+	}
+
+	result, err := anonymizer.secretsClient.Apply(context.TODO(), secret, applyOptions)
+	if err != nil {
+		klog.Errorf("Failed to create/update the translation table secret. err: %s", err)
+		return nil
+	}
+	klog.V(3).Infof("Created/Updated %s secret in %s namespace", TranslationTableSecretName, secretNamespace)
+	return result
+}
+
+// Resets the translation table, so that the translation table of multiple gathers wont mix toghater.
```

Comment:
They would actually mix now. Just tested by faking the translation table right before saving and got all the previous values in the secret and my fake ones.

Reply:
Nice catch, I replaced the `apply` with a `delete` and `create`, it should fix this issue.